### DISCUSSION TITLE
What's the default value of resolution in subquery?

### DISCUSSION THREAD

**Discussion by wangzhao765** [cortexproject/cortex#3630] [Q&A] [answered]
Example query:
`sum_over_time(irate(some_metric[20s])[2m:])`

(The interval of some_metric is 10s, and the step in query_range is 10s)

When I query the data from Prometheus, I can get a group of results.
But when I query the data from Cortex, I must modify the query to sum_over_time(irate(some_metric[20s])[2m:10s]) to get the same results. If I don't set the resolution of the subquery to 10s, from the results, it seems that the default value is 1m.

So what's the default value of resolution in subquery in Cortex? In Prometheus, it should be equal to the step in query_range query. https://prometheus.io/blog/2019/01/28/subquery-support/

**Comment by pracucci** [comment, marked as answer]
Cortex just uses the Prometheus PromQL engine, so there's no difference compared to Prometheus when it comes to the default resolution in a subquery.
